File: input/java/Outer01.java
class Outer01 {
    int num;

    // inner class
    private class Inner01 {
        public void print() {
            System.out.println("This is an inner class");
            System.out.println(num);
        }
    }

    // Accessing he inner class from the method within
    void displayInner() {
        Inner01 inner = new Inner01();
        inner.print();
    }

    void increase() {
        num = num + 1;
    }

    void decrease() {
        num = num - 1;
    }

    int getNum() {
        return num;
    }
}

File: input/kotlin/Outer01.kt
import Outer01.Inner01

internal class Outer01 {
    var num = 0

    // inner class
    private inner class Inner01 {
        fun print() {
            println("This is an inner class")
            println(num)
        }
    }

    // Accessing he inner class from the method within
    fun displayInner() {
        val inner = Inner01()
        inner.print()
    }

    fun increase() {
        num = num + 1
    }

    fun decrease() {
        num = num - 1
    }
}